engine-tests/src/utils/mod.rs

```diff
@@ -1029,10 +1029,10 @@ pub fn assert_gas_bound(total_gas: u64, tgas_bound: u64) {
 
 /// Returns true if `abs(a - b) / max(a, b) <= x / 100`. The implementation is written differently than
 /// this simpler formula to avoid floating point arithmetic.
-pub const fn within_x_percent(x: u64, a: u64, b: u64) -> bool {
+#[allow(clippy::cast_precision_loss, clippy::as_conversions)]
+pub fn within_x_percent(x: u64, a: u64, b: u64) -> bool {
     let (larger, smaller) = if a < b { (b, a) } else { (a, b) };
-
-    (100 / x) * (larger - smaller) <= larger
+    (100. / x as f64) * (larger as f64 - smaller as f64) <= larger as f64
```

Comment:
Why the change to `f64`?

Reply:
For `u64` example: `100/24 == 100/21`, it's too broad an assumption. ))

Reply:
Maybe it makes more sense to remove the division from the expression entirely and instead write the condition as

```
100 * (larger - smaller) <= larger * x
```

Reply:
Done: [ab48316](https://github.com/aurora-is-near/aurora-engine/pull/934/commits/ab483162d5434fd7a6647aa82639b31c120d0089)